File: frontend/types/smartTrim.ts
export type SmartTrimItem = {
  word: string;
  weight: number;
  value: number;
};

export type SmartTrimRequest = {
  text: string;
  limit: number;
};

export type SmartTrimResponse = {
  text: string;
  limit: number;
  items: SmartTrimItem[];
  total_value: number;
  total_weight: number;
  dp_table: number[][];
  success: boolean;
  error: string | null;
};

File: backend/app/api/schemas/smart_trim.py
from pydantic import BaseModel, Field

class SmartTrimRequest(BaseModel):
    text: str = Field(..., min_length=1, max_length=10000)
    limit: int = Field(..., ge=1, le=5000)

class SmartTrimItemResponse(BaseModel):
    word: str
    weight: int
    value: float

class SmartTrimResponse(BaseModel):
    text: str
    limit: int
    items: list[SmartTrimItemResponse]
    total_value: float
    total_weight: int
    dp_table: list[list[float]]
    success: bool
    error: str | None = None